olm/operator-registry.md

```diff
@@ -274,6 +285,37 @@ The on cluster workflow:
   - If the API returns null, OLM will ask for the bundle image URI from the operator-registry database and create a deployment with that image in order to get the manifests from that bundle. In order to do this, we will use an init container to push a binary into the container that can understand where the manifests are (in a well known path) and push all of those files into a configmap as strings.
   - OLM will read that configmap and create a CSV cr to drive the installation of the operator.
 
+
+#### CI story
+
+Not scoped out inside this enhancement is the fact that a higher order CLI tool needs to be built that encapsulates the functionality of these new `operator-registry` in order to build the index images. Such a tool will encapsulate the image build process along the lines of:
+
+1. Generate index scratch
+
+2. Generate index from previous index
+
+3. Delete entire operator from index
+
+4. Delete latest operator bundle version from index
+
+In all cases, such a tool will need to generate container images as output based on a set of input commands. Essentially it will generate a dockerfile/image specification, and then run podman/docker build to generate that image. Example dockerfile for an add command:
+
+```
+FROM quay.io/redhat/community-operators:latest as builder
+
+COPY exampledb.db exampledb.db
+
+FROM quay.io/operator-registry-builder/builder:latest
+COPY /build/bin/operator-registry /operator-registry
+
+RUN operator-registry add "quay.io/community-operators/foo:sha256@abcd123,quay.io/community-operators/bar:sha256@defg456" exampledb.db
```

Comment:
We have an open question/risk with this correct? We do not know if it is possible to pull an image during image creation. Can we be more clear that this process may need to change as we fill in gaps?

Reply:
@shawn-hurley I spoke with @ralphbean and he seemed to think that this process of pulling images should be okay for the current pipeline systems. I can make a note of the fact that we may need to provide some alternative solutions for those internal pipelines (building the database upfront then passing that db file around, for example) but that we are not promoting that as a primary workflow.